zyte_common_items/pages.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime
+
+import attrs
+from web_poet import ItemPage, RequestUrl, Returns, WebPage, field
+
+from .components import Metadata
+from .items import Product, ProductList
+
+
+class _BaseMixin:
+    def _get_response_url(self):
+        raise NotImplementedError
+
+    @field
+    def metadata(self) -> Metadata:
+        return Metadata(
+            dateDownloaded=f"{datetime.utcnow().isoformat(timespec='seconds')}Z",
+            probability=1.0,
+        )
+
+    @field
+    def url(self) -> str:
+        return str(self._get_response_url())
```

Comment:
What's the purpose of this indirection? The alternative is to add url field to BasePage and Page classes.